### PR TITLE
Enhance Notification theme for status/kind-specific `message` and `title` colors

### LINKED ISSUE #7410: Notification — Add ability to theme message/title color per “kind”
### Is this something you or your team is interested in contributing a pull request towards?
Yes.

### Expected Behavior

Ability to theme message/title color per kind/status.

### Actual Behavior

Only can define a single message/title color

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Enhances Notification theme to allow status/kind-specific `message` and `title` colors. This will us to leverage our token color system where there will be specific `on` colors for text colors on status backgrounds. We won't need to leverage all of these combinations (the background/text colors for our global and inline notifications are the same-- only toast differs), but introducing them for consistency.

#### Where should the reviewer start?
src/js/components/Notification/Notification.js

#### What testing has been done on this PR?
Locally, adjusted preview.js with the following:

```
import React, { useState, useEffect } from 'react';
import Root from 'react-shadow';
import { StyleSheetManager } from 'styled-components';
import { hpe } from 'grommet-theme-hpe';
import isChromatic from 'chromatic/isChromatic';
import { Grommet, grommet, hacktoberfest2022, Box, Text } from '../src/js';
import { deepMerge } from '../src/js/utils';

const CUSTOM_THEMED = 'Custom Themed';

const customTheme = deepMerge(hpe, {
  notification: {
    title: {
      color: 'blue',
    },
    message: {
      color: 'red',
    },
    normal: {
      message: {
        color: 'blue',
      },
      title: {
        color: 'red',
      },
      global: {
        title: {
          color: 'green',
        },
        message: {
          color: 'purple',
        },
      },
      toast: {
        title: {
          color: 'skyblue',
        },
        message: {
          color: 'green',
        },
      },
    },
  },
});

const THEMES = {
  hpe: customTheme,
  grommet,
  hacktoberfest2022,
  base: {},
};

export const decorators = [
  (Story, context) => {
    const [rootRef, setRootRef] = useState(null);
    const [state, setState] = useState('grommet');
    const [root, setRoot] = useState('document');
    const full = context.allArgs?.full || 'min';
    const dir = context.allArgs?.dir;
    const options = context.allArgs?.options;

    useEffect(() => {
      setState(context.globals.theme);
    }, [context.globals.theme]);
    useEffect(() => {
      setRoot(context.globals.root);
    }, [context.globals.root]);

    /**
     * This demonstrates that custom themed stories are driven off the "base"
     * theme. Custom themed stories will live under a "CustomThemed" directory.
     */
    if (context.kind.split('/')[2] === CUSTOM_THEMED && state !== 'base') {
      // if we are running the story in chromatic we want the chromatic snapshot
      // to be taken in the base theme for custom theme stories
      if (isChromatic()) {
        return (
          <Grommet theme={THEMES.base}>
            <Story state={THEMES.base} />
          </Grommet>
        );
      }
      return (
        <Box align="center" pad="large">
          <Text size="large">
            {`Custom themed stories are only displayed in the
                "base" theme mode. To enable, select "base" from the
                Theme menu above.`}
          </Text>
          <div hidden>
            <Story state={THEMES[state]} />
          </div>
        </Box>
      );
    }

    if (root === 'shadow') {
      return (
        // eslint-disable-next-line react/jsx-pascal-case
        <Root.div ref={setRootRef}>
          {rootRef && (
            <StyleSheetManager target={rootRef.shadowRoot}>
              <Grommet
                theme={THEMES[state]}
                full={full}
                dir={dir}
                options={options}
                containerTarget={rootRef.shadowRoot}
              >
                <Story state={THEMES[state]} />
              </Grommet>
            </StyleSheetManager>
          )}
        </Root.div>
      );
    }

    return (
      <Grommet theme={THEMES[state]} full={full} dir={dir} options={options}>
        <Story state={THEMES[state]} />
      </Grommet>
    );
  },
];

export const parameters = {
  layout: 'fullscreen',
  options: {
    storySort: (first, second) => {
      /**
       * The story sort algorithm will only ever compare two stories
       * a single time. This means that every story will only ever be either
       * the "first" parameter OR the "second" parameter, but not both.
       * So, the checks for custom themed stories need to happen on both inputs
       * of this function.
       *
       * A return value of 1 results in sorting the "first" story AFTER the
       * "second" story.
       *
       * A return value of 0 results in sorting the "first" story BEFORE the
       * secondary story.
       */
      const isFirstCustom = first[1].kind.split('/')[2] === CUSTOM_THEMED;
      const isSecondCustom = second[1].kind.split('/')[2] === CUSTOM_THEMED;
      if (isFirstCustom) return 1;
      if (isSecondCustom) return 0;
      return first[1].kind === second[1].kind
        ? 0
        : first[1].id.localeCompare(second[1].id, undefined, { numeric: true });
    },
  },
};

export const globalTypes = {
  theme: {
    defaultValue: 'grommet',
    toolbar: {
      title: 'Theme',
      items: [
        { title: 'base', value: 'base' },
        { title: 'grommet', value: 'grommet' },
        { title: 'hpe', value: 'hpe' },
        { title: 'hacktoberfest2022', value: 'hacktoberfest2022' },
      ],
    },
  },
  root: {
    defaultValue: 'document',
    toolbar: {
      title: 'Root',
      items: ['document', 'shadow'],
    },
  },
};
```

Screenshot 2024-10-31 at 2 06 35 PM

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Closes #7410 

#### Do the grommet docs need to be updated?
Yes.

For each status and each kind
- [ ] `theme.notification.[status].[kind].message.color
- [ ] `theme.notification.[status].[kind].title.color

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 1]
```diff
@@ -339,4 +339,64 @@ describe('Notification', () => {
...
+      notification: {
+        title: {
+          color: 'blue',
+        },
+        message: {
+          color: 'red',
+        },
+        normal: {
+          message: {
+            color: 'blue',
```
Comment (britt6612) [inline src/js/components/Notification/__tests__/Notification-test.tsx:354]: these colors arent coming thru correctly

Comment (britt6612) [reply src/js/components/Notification/__tests__/Notification-test.tsx:354]: ignore I was on the wrong branch 😆

File: src/js/themes/base.js [hunk 2]
```diff
@@ -1274,9 +1274,21 @@ export const generate = (baseSpacing = 24, scale = 6) => {
           opacity: 'weak',
         },
         color: 'status-critical',
+        // message: {
```
Comment (britt6612) [inline src/js/themes/base.js:1277]: do we also need 
```
        // message: {
        //   color: undefined,
        // },
        // title: {
        //   color: undefined,
        // },
```
above under just `notification.title` ? I see you have it under all the `critical` `warning` `normal` etc but I think we need it above as well?

Comment (britt6612) [reply src/js/themes/base.js:1277]: do we also need under `notification.global`?

Comment (taysea) [reply src/js/themes/base.js:1277]: The naming structure is intended to be `theme.notification.[status].[kind].message` -- so it's not needed under `notification.global`

Comment (britt6612) [reply src/js/themes/base.js:1277]: aw yes you are right thought I saw this in the `base.d.ts` file but thanks for clarifying

**Review by britt6612 — APPROVED** [review]

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**